lib/src/main/java/com/futurewei/alcor/common/db/CacheFactory.java

```diff
@@ -43,27 +49,46 @@ CacheFactory cacheFactoryInstance() {
         return new CacheFactory();
     }
 
-    private ICache getIgniteCache(String cacheName) {
+    private <K, V> ICache<K, V> getIgniteCache(String cacheName) {
         return new IgniteCache<>(igniteClient, cacheName);
     }
 
-    public <K, V> ICache getRedisCache(Class<V> v, String cacheName) {
-        RedisTemplate<K, V> template = new RedisTemplate<K, V>();
+    public <K, V> RedisTemplate<K, V> getRedisTemplate(Class<V> v){
```

Comment:
As this is a general interface for Ignite, Redis and other DBs, ideally we should keep the interface db-independent. Could we move this method to Redis class only? 

@Gzure @chenpiaoping We need to add a comment to this interface. Thank you.

Reply:
I think we could create a CacheFactory interface, and a DB interface implement a CacheFactory. So we will return different DB CacheFactory from application properties. We can make all DB keep independent.